File: VisionVibe BE/VisionVibe/src/main/java/com/cx/visionvibe/entity/Order.java
package com.cx.visionvibe.entity;

import com.cx.visionvibe.enums.OrderStatus;
import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;
import org.springframework.data.annotation.CreatedDate;

import java.time.LocalDateTime;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "orders")
@FieldDefaults(level = AccessLevel.PRIVATE)
public class Order {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long id;
    
    @Column(name = "customer_name", nullable = false)
    String customerName;

    @Column(nullable = false)
    String email;

    @Column(nullable = false)
    String address;

    @Column(nullable = false)
    String phone;

    @Enumerated(EnumType.STRING)
    OrderStatus orderStatus;

    String note;

    @Column(name = "created_at")
    LocalDateTime createdAt;

    @OneToMany(mappedBy = "order", cascade = CascadeType.ALL)
    List<Notification> notifications;

    @OneToMany(mappedBy = "order", cascade = CascadeType.ALL)
    List<OrderDetail> orderDetails;

    @ManyToOne
    @JoinColumn(name = "payment_method_id")
    PaymentMethod paymentMethod;
}

File: VisionVibe-FE/src/app/dto/Order.ts
import { ProductDetail } from "./ProductDetail";

export interface Order {
    id: number;
    customerName: string;
    email: string;
    phoneNumber: string;
    address: string;
    note: string;
    createdAt: Date;
    status: string;
    paymentMethodResponse: PaymentMethod;
    paymentUrl: string | null;
    orderDetailResponseList: OrderDetail[];
}

export interface PaymentMethod {
    id: number;
    name: string;
}

export interface OrderDetail {
    id: number;
    productDetailResponse: ProductDetail;
    amount: number;
}